oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedMongoDownloadTask.java

```diff
@@ -682,8 +677,15 @@ private void download(Bson mongoQueryFilter) throws InterruptedException, Timeou
                             downloadRange(nextRange, mongoQueryFilter, downloadOrder);
                         }
                         downloadCompleted = true;
+                    } catch (IllegalStateException | InterruptedException | MongoInterruptedException e) {
+                        if (cancelled) {
```

Comment:
Same here with the interrupt flag, https://www.baeldung.com/java-interrupted-exception

Reply:
Here we have handled the interrupt by stopping the task, so I don't think we need to propagate the interrupt.